src/classes/BDI_Donations.cls

```diff
@@ -236,22 +244,55 @@ public class BDI_Donations {
 
         // add any new payments to our map
         trackPayments(dataImportDonations);
-        
+
         // when we mark Payments paid, ensure they will close our Opp.
         npe01__Contacts_And_Orgs_Settings__c cs = UTIL_CustomSettingsFacade.getContactsSettings();
         if (cs.Payments_Auto_Close_Stage_Name__c == null) {
             cs.Payments_Auto_Close_Stage_Name__c = defaultClosedWonStage;
         }
-        
+
         // update payments with any data from the DI record
         updatePayments(dataImportDonations);
     }
 
+    /*******************************************************************************************************
+    * @description Determine if the data import batch number should be set on the opportunity or payment
+    * @param targetObject The opportunity or payment SObject record
+     * @param dataImport The data import record being processed
+    * @return Boolean Whether or not the DI batch number should be set on opportunities or payments
+    */
+    private Boolean shouldSetDonationBatchNumber(SObject targetObject, DataImport__c dataImport) {
+        SObjectType objectType = targetObject.getSObjectType();
+
+        if (bdi.isDryRun ||
+            String.isBlank(dataImport.NPSP_Data_Import_Batch__r.Batch_Number__c)) {
+
+            return false;
+        }
+
+        if (objectType == Opportunity.SObjectType) {
+            return targetObject.get(Opportunity.Id) == null;
+        } else if (objectType == npe01__OppPayment__c.SObjectType) {
+            return targetObject.get(npe01__OppPayment__c.npe01__Paid__c) == true;
+        }
+        return false;
+    }
+
+    /*******************************************************************************************************
+    * @description Sets the batch number onto the target SObject record
+    * @param targetObject The opportunity or payment SObject record
+     * @param field The target field to populate
+     * @param value The target value to set in the target field
+    */
+    private void setBatchNumber(SObject targetObject, SObjectField field, String value) {
+        targetObject.put(field, value);
+    }
```

Comment:
This method is called "setBatchNumber" but is replicating the built-in SObject put() method, which is generic and handles setting the value of any field on the object.  Is there a reason for using this method vs. setting the field values in-line at the call site?

Reply:
This is a good point. I had originally thought I would use that method in a different way before realizing the SObject could be updated in memory. So I agree that this method probably isn't needed anymore. Will update it and remove it and use the put() inline

Reply:
👍🏽